### DISCUSSION TITLE
UPDATE Server CA  certificates

### DISCUSSION THREAD

**Discussion by KiwiActinidia** [dani-garcia/vaultwarden#3319] [Q&A]
### Subject of the issue

SMTP Server connection error

* vaultwarden version: 1.27.0

* Install method: docker-compose

* Clients used:  web 

* Reverse proxy and version: apache2  

* MySQL/MariaDB or PostgreSQL version: 

* Other relevant details:

### Steps to reproduce

connect to a SMTP server using latest Let's Encrypt certificates

### Expected behaviour

No error sending email

### Actual behaviour

certificates error

Setting SMTP_ACCEPT_INVALID_CERTS=true
no problem sending email

### Troubleshooting data

vaultwarden  | [2023-03-07 14:15:56.592][vaultwarden::mail][ERROR] SMTP Connection error: Connection error: error:1416F086:SSL routines:tls_process_server_certificate:certificate verify failed:../ssl/statem/statem_clnt.c:1914: (unable to get local issuer certificate)
vaultwarden  | [2023-03-07 14:15:56.592][vaultwarden::api::core::accounts][ERROR] Error sending welcome email: SMTP Connection error: Connection error: error:1416F086:SSL routines:tls_process_server_certificate:certificate verify failed:../ssl/statem/statem_clnt.c:1914: (unable to get local issuer certificate) 

Regards
Alessandro

**Comment by BlackDex** [comment]
What kind of certificate are you using? Are you providing the full chain and not only the top-level certificate?
Because, I have the exact same setup, and that works just fine for me.

I suggest to use [testssl.sh](https://github.com/drwetter/testssl.sh) and run it like this to test:
```bash
git clone https://github.com/drwetter/testssl.sh.git
cd testssl.sh
./testssl.sh --starttls smtp mail.domain.tld:25
```

And then look for the `Certificates provided`, `Issuer` and `Intermediate cert validity` values.
If that does not provide more then 1 cert, then you are not using the correct chained certificate file on your mail server.

**Comment by KiwiActinidia** [comment]
Your suggestion very useful. I found the problem and discover a new SSL tools.
Regards
Alessandro
